src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -124,17 +123,20 @@ def generate_initial_population(self, pop_data: SimulantData) -> None:
         pop["date_of_birth"] = self.start_time - pd.to_timedelta(
             np.round(pop["age"] * 365.25), unit="days"
         )
-
         # format
-        n_chosen = pop.shape[0]
+        n_chosen = pop.shape[0]  # what is this?
+
+       # Add Social Security Numbers
         pop["ssn"] = self.ssn_generator.generate(pop).ssn
+        # Check for duplicate ssn
+        while len(pop.ssn) != len(pop.ssn.unique()):
```

Comment:
if we have a really huge population, what are the chances that this starts to have to loop many many times?

^ if that's a possibility, we could switch to looping over regenerating just the duplicates. im not sure if that ends up saving a lot of time or no, but possibly(?) if we have 350 million sims

we do this a couple places in the code. for example in `persons.py`:

```
    def _get_new_household_ids(
        self, persons_who_move: pd.DataFrame, event: Event
    ) -> pd.Series:
        households = self.population_view.subview(["household_id"]).get(event.index)
        all_household_ids = list(households.squeeze().drop_duplicates())

        new_household_ids = persons_who_move["household_id"].copy()
        additional_seed = 0
        while (new_household_ids == persons_who_move.household_id).any():
            unchanged_households = new_household_ids == persons_who_move.household_id
            new_household_ids[unchanged_households] = self.randomness.choice(
                new_household_ids.loc[unchanged_households].index,
                all_household_ids,
                additional_key=additional_seed,
            )
            additional_seed += 1

        return pd.Series(new_household_ids)
```